### PR TITLE
python(feat): Add units_row and descriptions_row option to simpele cs…

### PR DESCRIPTION
**Pull request by marcsiftstack**
Some customers ad extra header rows with units. Also allow a row with descriptions.

This PR also adds logic to detect if an integer is a unit64 instead of assuming every integer is an int.

### CODE REVIEW

File: python/lib/sift_py/data_import/csv.py [hunk 1]
```diff
@@ -149,17 +152,45 @@ def simple_upload(
...
+
+        def is_uint64(n: pd.Series) -> bool:
+            int64_max = 2**63 - 1
+            return bool((n > int64_max).any())
+
         data_config = {}
-        df = pd.read_csv(path)
+        df = pd.read_csv(path, skiprows=skip_rows)
+
+        units: Optional[List[str]] = None
```
Comment (solidiquis) [inline python/lib/sift_py/data_import/csv.py:184]: Can this just be initialized as an empty list? 
```
>>> bool([])
False
>>> bool(None)
False
```

File: python/lib/sift_py/data_import/csv.py [hunk 2]
```diff
@@ -149,17 +152,45 @@ def simple_upload(
...
+        if units_row is not None:
+            df_units = pd.read_csv(path, nrows=units_row)
+            units = cast(List[str], df_units.iloc[units_row - 1].astype(str))
+
+        descriptions: Optional[List[str]] = None
```
Comment (solidiquis) [inline python/lib/sift_py/data_import/csv.py:189]: same here

**Review by solidiquis — CHANGES_REQUESTED** [review]
small nits

File: python/lib/sift_py/data_import/csv.py [hunk 3]
```diff
@@ -181,12 +181,12 @@ def is_uint64(n: pd.Series) -> bool:
...
+        units: List[str] = []
         if units_row is not None:
             df_units = pd.read_csv(path, nrows=units_row)
             units = cast(List[str], df_units.iloc[units_row - 1].astype(str))
 
-        descriptions: Optional[List[str]] = None
+        descriptions: List[str] = []
         if descriptions_row is not None:
             df_descriptions = pd.read_csv(path, nrows=descriptions_row)
             descriptions = cast(List[str], df_descriptions.iloc[descriptions_row - 1].astype(str))
```
Comment (solidiquis) [inline python/lib/sift_py/data_import/csv.py:192]: dont think we need these casts anymore either

File: python/lib/sift_py/data_import/csv.py [hunk 4]
```diff
@@ -181,12 +181,12 @@ def is_uint64(n: pd.Series) -> bool:
         data_config = {}
         df = pd.read_csv(path, skiprows=skip_rows)
 
-        units: Optional[List[str]] = None
+        units: List[str] = []
         if units_row is not None:
             df_units = pd.read_csv(path, nrows=units_row)
             units = cast(List[str], df_units.iloc[units_row - 1].astype(str))
```
Comment (solidiquis) [inline python/lib/sift_py/data_import/csv.py:187]: same here

**Review by solidiquis — CHANGES_REQUESTED** [review]

File: python/lib/sift_py/data_import/csv.py [hunk 5]
```diff
@@ -170,10 +203,20 @@ def simple_upload(
                 raise Exception(
                     f"Unable to upload data type in column {i+1} {header}. Inferred type: {inferred_dtype}"
                 )
+            if dtype == "int":
```
Comment (niliayu) [inline python/lib/sift_py/data_import/csv.py:206]: What's your thinking for using `infer_dtype()` and doing this vs using `df.dtypes` for getting the datatype for each column?

Comment (marcsiftstack) [reply python/lib/sift_py/data_import/csv.py:206]: No real reason, this was just the first way I tried.

Comment (niliayu) [reply python/lib/sift_py/data_import/csv.py:206]: Gotcha! Would you be willing to try using `df.dtypes` to see if that would simplify things here? I wonder if we wouldn't need the `is_uint64` check and could save a few lines there.

File: python/lib/sift_py/data_import/csv.py (hunk 3 above)
Comment (marcsiftstack) [reply python/lib/sift_py/data_import/csv.py]: Mypy complains because astype is a `Series[bool]`.

File: python/lib/sift_py/data_import/csv.py [hunk 6]
```diff
@@ -197,16 +187,20 @@ def is_uint64(n: pd.Series) -> bool:
             if i + 1 == time_column:
                 continue
 
-            inferred_dtype = pd.api.types.infer_dtype(df[df.columns[i]], skipna=False)
-            dtype = types.get(inferred_dtype)
-            if dtype is None:
+            raw_dtype = str(df[df.columns[i]].dtype)
+            if raw_dtype == "float64":
+                raw_dtype = "double"
```
Comment (niliayu) [inline python/lib/sift_py/data_import/csv.py:192]: Why change this to `double`?

Comment (marcsiftstack) [reply python/lib/sift_py/data_import/csv.py:192]: https://github.com/sift-stack/sift/blob/python/ch10_upload_service/python/lib/sift_py/ingestion/channel.py#L187

**Review by niliayu — APPROVED** [review]
lgtm!

**Review by solidiquis — APPROVED** [review]

**Pull request merged**